packages/memory-inspector/src/browser/editable-widget/memory-editable-table-widget.tsx

```diff
@@ -0,0 +1,353 @@
+export namespace EditableMemoryWidget {
+    export const ID = 'editable.memory.widget';
+}
+
+@injectable()
+export class MemoryEditableTableWidget extends MemoryTableWidget {
+    protected pendingMemoryEdits = new Map<string, string>();
+    protected override previousBytes: Interfaces.LabeledUint8Array | undefined;
+    protected memoryEditsCompleted = new Deferred<void>();
+    protected highlightedField: Long = Long.fromInt(-1);
+    protected writeErrorInfo: { location: string, error: string } | undefined;
+    protected currentErrorTimeout: number | undefined;
+
+    protected doShowMoreMemoryBefore = false;
+    protected doShowMoreMemoryAfter = false;
+
+    @postConstruct()
+    protected override async init(): Promise<void> {
+        this.memoryEditsCompleted.resolve();
+        await super.init();
+        this.addClass('editable');
+    }
+
+    resetModifiedValue(valueAddress: Long): void {
+        const didChange = this.pendingMemoryEdits.delete(valueAddress.toString());
+        if (didChange) {
+            this.update();
+        }
+    }
+
+    protected override getState(): void {
+        super.getState();
+        if (!this.isInBounds(this.highlightedField)) {
+            this.highlightedField = this.memory.address;
+        }
+    }
+
+    protected override async handleMemoryChange(newMemory: Interfaces.MemoryReadResult): Promise<void> {
+        await this.memoryEditsCompleted.promise;
+        super.handleMemoryChange(newMemory);
+    }
+
+    protected override areSameRegion(a: Interfaces.MemoryReadResult, b?: Interfaces.MemoryReadResult): boolean {
+        return b !== undefined && a.address.equals(b.address) && a.bytes.length === b.bytes.length;
+    }
+
+    protected override getTableFooter(): React.ReactNode {
+        const showButtons = !!this.pendingMemoryEdits.size;
+        return (
+            (showButtons || this.writeErrorInfo) && (
+                <div className='memory-edit-button-container'>
+                    {showButtons && <button
+                        className='theia-button secondary'
+                        onClick={this.handleClearEditClick}
+                        type='reset'
+                        title={nls.localize('theia/memory-inspector/editable/clear', 'Clear Changes')}
+                    >
+                        {nls.localize('theia/memory-inspector/editable/clear', 'Clear Changes')}
+                    </button>}
+                    {showButtons && <button
+                        className='theia-button main'
+                        onClick={this.submitMemoryEdits}
+                        type='submit'
+                        title={nls.localize('theia/memory-inspector/editable/apply', 'Apply Changes')}
+                    >
+                        {nls.localize('theia/memory-inspector/editable/apply', 'Apply Changes')}
+                    </button>}
+                    {!!this.writeErrorInfo && <div className='memory-edit-error'>
```

Comment:
The way this element is styled causes some unexpected rendering for me:

![image](https://user-images.githubusercontent.com/19170971/186367798-5264bf39-a111-40a2-9f4b-cd2b0e29d301.png)

Due to the absolute positioning, the error div only "hides" the buttons but also overlays the 'Load...' message for me.  Do you see the same behavior? I think we have two options here:
- Show the error simply below the buttons
- Only render the buttons when there is no error or fix the overlay by ensuring that it does not overlap with the 'Load...' message. The easiest option for me was to change the style from `bottom: 0` to `top: 0`.
 
But maybe this is just caused by my setup and it can be easily fixed with CSS so it should not be a blocker if it looks good for you.

Reply:
I've adjusted this a bit. It will still appear to occlude the 'load more memory' area (because the error takes up more room, so the bottom of the widget is moved up), but now it will be possible to scroll down a little further to reveal the 'load more memory' option again.

Reply:
Looks good to me, thank you for taking another look at it.